controllers/designate_controller.go

```diff
@@ -713,6 +714,99 @@ func (r *DesignateReconciler) reconcileNormal(ctx context.Context, instance *des
 	}
 	Log.Info("Deployment API task reconciled")
 
+	// Handle Mdns predictable IPs configmap
+	nad, err := nad.GetNADWithName(ctx, helper, instance.Spec.DesignateNetworkAttachment, instance.Namespace)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	networkParameters, err := designate.GetNetworkParametersFromNAD(nad)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	//
+	// Predictable IPs.
+	//
+	// NOTE(oschwart): refactoring this might be nice. This could also  be
+	// optimized but the data sets are small (nodes an IP ranges are less than
+	// 100) so optimization might be a waste.
+	//
+	predictableIPParams, err := designate.GetPredictableIPAM(networkParameters)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	// Fetch Mdns config map
+	mdnsLabels := labels.GetLabels(instance, labels.GetGroupLabel(instance.ObjectMeta.Name), map[string]string{})
+	nodeConfigMap, err := r.handleConfigMap(ctx, helper, instance, designate.MdnsPredIPConfigMap, mdnsLabels)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+	// Get a list of the nodes in the cluster
+
+	// TODO(oschwart):
+	// * confirm whether or not this lists only the nodes we want (i.e. ones
+	// that will host the daemonset)
+	// * do we want to provide a mechanism to temporarily disabling this list
+	// for maintenance windows where nodes might be "coming and going"
+
+	nodes, err := helper.GetKClient().CoreV1().Nodes().List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	var nodeNames []string
+	for _, node := range nodes.Items {
+		nodeNames = append(nodeNames, fmt.Sprintf("mdns_%s", node.Name))
+	}
+
+	allocatedIPs := make(map[string]bool)
+	updatedMap, allocatedIPs, err := r.allocatePredictableIPs(ctx, predictableIPParams, nodeNames, nodeConfigMap.Data, allocatedIPs)
```

Comment:
IIUC this will work the first time around, but if the number of nodes for minidns nodes runs, you are going to run into collisions with the IPs allocated for the bind hosts. What you need to do is get ALL of the allocated IPs for both config maps first.

Reply:
Thanks, I have just added fetching both configmaps before I allocate predictable IPs.